backend/plugin/db/driver.go

```diff
@@ -240,6 +240,9 @@ type ExecuteOptions struct {
 
 	// The maximum number of retries for lock timeout statements.
 	MaximumRetries int
+	
+	// Task type for determining default transaction mode
+	TaskType storepb.Task_Type
```

Comment:
TaskType does not belong to ExecuteOptions. Lets do something like
```suggestion
	// transaction mode
	WrapInTx bool
```